sparksql-scalapb/src/main/scala/scalapb/spark/SchemaOptions.scala

```diff
@@ -1,41 +1,37 @@
 package scalapb.spark
 
-import scalapb.descriptors.FieldDescriptor
-import scalapb.GeneratedMessage
-import scalapb.GeneratedMessageCompanion
-import com.google.protobuf.Field
-import org.apache.parquet.example.data.simple.Primitive
-import org.apache.arrow.flatbuf.Schema
-import scalapb.descriptors.Descriptor
-import org.apache.spark.sql.types.DataType
-import org.apache.spark.sql.types.DoubleType
-import org.apache.spark.sql.types.BooleanType
-import org.apache.spark.sql.types.IntegerType
-import org.apache.spark.sql.types.LongType
-import org.apache.spark.sql.types.StringType
-import org.apache.spark.sql.types.FloatType
-import org.apache.spark.sql.types.BinaryType
+import org.apache.spark.sql.types._
+import scalapb.{GeneratedMessage, GeneratedMessageCompanion}
+import scalapb.descriptors.{Descriptor, FieldDescriptor}
 
 case class SchemaOptions(
     columnNaming: ColumnNaming,
-    retainPrimitiveWrappers: Boolean
+    retainPrimitiveWrappers: Boolean,
+    catalystMappers: Seq[CatalystMapper]
```

Comment:
I prefer this would be `Map[GeneratedMessageCompanion[_], TypedEncoder[_]]`? - it would be more consistent with other areas of ScalaPB (such as json4s), simplify the lookup logic that is repeated in this PR, and let us rely on a type from frameless that already has a few specific instances users can leverage.